### ISSUE TITLE
tests: kernel: context: seems to be failing on Nordic platforms

### ISSUE THREAD

**Issue by ioannisg** [zephyrproject-rtos/zephyr#25194]
**Describe the bug**
tests/kernel/context seems to be failing on several Nordic nRF Platforms

**To Reproduce**
1. mkdir build; cd build
2. cmake -DBOARD=board\nrf52dk_nrf52832
3. make

**Expected behavior**
Should pass

**Screenshots or console output**
```
*** Booting Zephyr OS build v2.3.0-rc1-9-gb88d79846641  ***
Running test suite context
===================================================================
starting test - test_kernel_interrupts
PASS - test_kernel_interrupts
===================================================================
starting test - test_kernel_timer_interrupts
SKIP - test_kernel_timer_interrupts
===================================================================
starting test - test_kernel_ctx_thread
Testing k_current_get() from an ISR and thread
Testing k_is_in_isr() from an ISR
Testing k_is_in_isr() from a preemptible thread
PASS - test_kernel_ctx_thread
===================================================================
starting test - test_busy_wait
Thread busy waiting for 20000 usecs
Thread busy waiting completed
Thread busy waiting for 20000 usecs (irqs locked)
Thread busy waiting completed (irqs locked)

    Assertion failed at ../src/main.c:786: test_busy_wait: (rv is true)
 *** thread timed out waiting for k_busy_wait()
FAIL - test_busy_wait
===================================================================
starting test - test_k_sleep
Testing k_thread_create() without cancellation
 thread (q order: 0, t/o: 1000) is running
E: ***** BUS FAULT *****
E:   Precise data bus error
E:   BFAR Address: 0xfffffff5
E: r0/a1:  0xfffffff5  r1/a2:  0xe000ed00  r2/a3:  0x20000b00
E: r3/a4:  0x20000438 r12/ip:  0x00000000 r14/lr:  0x00005897
E:  xpsr:  0x61000000
E: Faulting instruction address (r15/pc): 0x00005896
E: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
E: Current thread: 0x20000438 (unknown)
E: Halting system

```
**Environment (please complete the following information):**
 - Commit SHA or Version used. v2.3.0 rc1

**Comment by ioannisg** [comment]
On nRF5340 the error appears to differ:

```
starting test - test_busy_wait
Thread busy waiting for 20000 usecs
Thread busy waiting completed
Thread busy waiting for 20000 usecs (irqs locked)
Thread busy waiting completed (irqs locked)

    Assertion failed at ../src/main.c:786: test_busy_wait: (rv is true)
 *** thread timed out waiting for k_busy_wait()
FAIL - test_busy_wait
===================================================================
starting test - test_k_sleep
Testing k_thread_create() without cancellation
 thread (q order: 0, t/o: 1000) is running
E: ***** USAGE FAULT *****
E:   Unaligned memory access
E: r0/a1:  0xfffffff5  r1/a2:  0xe000ed00  r2/a3:  0x20000a90
E: r3/a4:  0x20000440 r12/ip:  0x00000000 r14/lr:  0x00005787
E:  xpsr:  0x61000000
E: Faulting instruction address (r15/pc): 0x00005786
E: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
E: Current thread: 0x20000440 (unknown)
E: Halting system

```

**Comment by Nukersson** [comment]
Seems to be already reported [here](https://github.com/zephyrproject-rtos/zephyr/issues/25176#issue-615313752)

**Comment by ioannisg** [comment]
Yeap, just saw it, and triaged that one, too.

**Comment by ioannisg** [comment]
duplicate of #25176

**Comment by stephanosio** [comment]
Fixed by #25203
